### PR TITLE
dont allow SentryDiagnosticListenerIntegration to be added multiple times

### LINKED ISSUE #1707: DiagnosticSource can easily be added twice
### Package

Sentry.DiagnosticSource

### .NET Flavor

.NET

### .NET Version

all

### OS

Other

### SDK Version

all

### Self-Hosted Sentry Version

_No response_

### Steps to Reproduce

Install Sentry.AspNet

Call AddAspNet and AddDiagnosticSourceIntegration

### Expected Result

Only add it once. 

### Actual Result

The integration is added twice.

https://github.com/getsentry/sentry-dotnet/blob/89cad6dd25fd9cb0df4051c7414fba8263b4f1ac/src/Sentry.DiagnosticSource/SentryOptionsDiagnosticExtensions.cs#L18

### PR DESCRIPTION
**Pull request by SimonCropp**
fixes #1707

### CODE REVIEW

File: src/Sentry.DiagnosticSource/SentryOptionsDiagnosticExtensions.cs [hunk 1]
```diff
@@ -15,7 +16,16 @@ public static class SentryOptionsDiagnosticExtensions
         /// </summary>
         /// <param name="options">The Sentry options.</param>
         public static void AddDiagnosticSourceIntegration(this SentryOptions options)
-            => options.AddIntegration(new SentryDiagnosticListenerIntegration());
+        {
+            if (options.Integrations != null &&
+                options.Integrations.OfType<SentryDiagnosticListenerIntegration>().Any())
```
Comment (SimonCropp) [inline src/Sentry.DiagnosticSource/SentryOptionsDiagnosticExtensions.cs:21]: @bruno-garcia @mattjohnsonpint  should i refactor this into a helper method in options?

File: src/Sentry.DiagnosticSource/SentryOptionsDiagnosticExtensions.cs [hunk 2]
```diff
@@ -15,7 +16,16 @@ public static class SentryOptionsDiagnosticExtensions
...
+            {
+                options.LogInfo($"{nameof(SentryDiagnosticListenerIntegration)} has already been added. The second call to {nameof(AddDiagnosticSourceIntegration)} will be ignored.");
```
Comment (mattjohnsonpint) [inline src/Sentry.DiagnosticSource/SentryOptionsDiagnosticExtensions.cs:23]:
```suggestion
                options.LogWarning($"{nameof(SentryDiagnosticListenerIntegration)} has already been added. The second call to {nameof(AddDiagnosticSourceIntegration)} will be ignored.");
```

File: src/Sentry.DiagnosticSource/SentryOptionsDiagnosticExtensions.cs (hunk 1 above)
Comment (mattjohnsonpint) [inline src/Sentry.DiagnosticSource/SentryOptionsDiagnosticExtensions.cs:21]:
```suggestion
            if (options.Integrations?.OfType<SentryDiagnosticListenerIntegration>().Any() is true)
```

Comment (mattjohnsonpint) [reply src/Sentry.DiagnosticSource/SentryOptionsDiagnosticExtensions.cs:21]: I don't think that's necessary.

File: src/Sentry/SentryOptionsExtensions.cs [hunk 3]
```diff
@@ -87,9 +87,16 @@ public static void DisableNetFxInstallationsIntegration(this SentryOptions optio
...
-                ? options.Integrations.Concat(new[] { integration }).ToArray()
-                : new[] { integration };
+        {
+            if (options.Integrations == null)
+            {
+                options.Integrations = new[] {integration};
+            }
+            else
+            {
+                options.Integrations = options.Integrations.Concat(new[] {integration}).ToArray();
```
Comment (mattjohnsonpint) [inline src/Sentry/SentryOptionsExtensions.cs:97]: Just a thought, but maybe we should leave `AddDiagnosticSourceIntegration` as it was, and check here more generally instead?   I mean, can we safely say that there should not be more than one integration of the same type?

Comment (SimonCropp) [reply src/Sentry/SentryOptionsExtensions.cs:97]: @mattjohnsonpint i asked @bruno-garcia the same question. he said it is valid to have more than one integration by type, but with different settings in each integration instance

Comment (bruno-garcia) [reply src/Sentry/SentryOptionsExtensions.cs:97]: Where did that communication happen?

The example was: `new FileWatcherIntegration("path1"), new FileWatcherIntegratoin("path2")`

Comment (mattjohnsonpint) [reply src/Sentry/SentryOptionsExtensions.cs:97]: LGTM

**Review by mattjohnsonpint — APPROVED** [review]

**Pull request merged**